### ISSUE TITLE
使用自己的数据集训练时，model loss 一直为1.0000？

### ISSUE THREAD

**Issue by lixingwei1106** [liuheng92/tensorflow_PSENet#34]
![2019-06-25 09-40-07屏幕截图](https://user-images.githubusercontent.com/33373430/60064339-ad1c9000-9732-11e9-8387-cc6a57255554.png)

相关信息及问题描述：
(1) 标签文件的格式为：116,1179,206,1179,206,1207,116,1207,"###"，保存为.txt文件；
(2) 标签文件和图片存放在同一个文件夹中；
(3) 模型可以正常训练，model loss从训练开始到结束一直为1.0000；total loss 的值最开始大于 
      1.0000，随着训练的进行，逐渐减小，直至total loss值变为1.0000，当total loss变为1后就不
      再发生变化。即训练结束时，model loss和total loss 的值均为1。(上图是训练初期的训练日志
      信息)

        使用训练好的模型测试时，无法检测文本，具体情况是输入图片和输出图片完全相同，类似于模型未对输入图片作任何处理，只是原样输出。请问，出现这个问题的原因是什么，我应该从哪些方面着手来解决该问题？

**Comment by liuheng92** [comment]
看看summary里面的图片输出是否正常

**Comment by lixingwei1106** [comment]
> 看看summary里面的图片输出是否正常

summary里面的图片指的是最终的检测结果吗？最终的输出图片没有异常，只是没有在文本上显示检测框

**Comment by piernikowyludek** [comment]
Hi, I had the same problem - model loss stuck at 1.0000 on my own dataset (quite different from ICDAR). What fixed it for me is just setting the training mask all to 1s.

From what I understand - areas set to 0.0 in the training mask prevent the error propagating from those areas. We essentially ignore certain areas for loss computation. 
In data_provider we find:

`cv2.fillPoly(training_mask, poly.astype(np.int32)[np.newaxis, :, :], 0)`

We set the inside of our bounding box predictions to 0? Why? Changing this 'unstuck' my training.

**Comment by dsandii** [comment]
@piernikowyludek Exactly which block did you change? I'm experiencing the same problem. I applied your change to both of the calls to `cv2.fillPoly()` in the `generate_seg()` function. This allowed my total loss to finally start to drop, but my model loss is still at 1.0. But thank you for sharing your solution.

**Comment by dsandii** [comment]
I resolved this problem by doing something else. When loading annotations, all of my tags were labeled `True`. If your tags are being labeled `True` the generator thinks it should ignore those tags. This was causing the loss to always be `1.0` for me.

**Comment by zonasw** [comment]
The reason for the error is that u tag contains “###”, in fact, u should not include "###", "*" and "?", here's why:
[if label == '*' or label == '###' or label == '?':    text_tags.append(True)](https://github.com/liuheng92/tensorflow_PSENet/blob/5d057c6eccb79900b60fa349b4dddc756ec61052/utils/data_provider/data_provider.py#L71)

u can change the “###” to 000 or whatever

**Comment by Leedorado** [comment]
how do you prepare your own dataset?Could you please show me a sample?thx

**Comment by jdmking2019** [comment]
请问下 原因找到了嘛 我也遇到这样的问题。
![image](https://user-images.githubusercontent.com/54650688/136875139-85f299be-1465-4eed-b657-b17c4ab6de11.png)
